### ISSUE TITLE
New parser handling of incomplete documents

### ISSUE THREAD

**Issue by ravangen** [rmosolgo/graphql-ruby#4746]
**Describe the bug**

With the new default parser, how should incomplete documents be handled and reported?

**Versions**

Ruby: `3.2.1`
`graphql`: `2.2.1` 

**Original `2.1.8` behaviour**

```ruby
> query = GraphQL::Query.new(TestSchema, "{")
=> query ...
> query.static_errors
=> [#<GraphQL::ParseError: Unexpected end of document>]
> query.result.to_h
=> {"errors"=>[{"message"=>"Unexpected end of document", "locations"=>[]}]}
```

**New `2.2.1` behaviour**

```ruby
> query = GraphQL::Query.new(TestSchema, "{")
=> query ...
> query.static_errors
=> []
> query.result.to_h
=> {}
```

**Additional context**

Is this by design? I expected a `GraphQL::ParseError: Unexpected end of document`

**Comment by myronmarston** [comment]
I noticed this and it surprised me, too.

**Comment by rmosolgo** [comment]
It _should_ have stayed the same! I thought it would have been covered by 

https://github.com/rmosolgo/graphql-ruby/blob/becd60008606f5ca750e568b3367f27b464d0fd6/spec/graphql/query_spec.rb#L655-L661

But apparently not...

**Comment by rmosolgo** [comment]
Fix in the works at #4749 -- it returns a _different_ error, but a slightly better one in my opinion.
